pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -43,15 +45,17 @@ type cephStatusChecker struct {
 	clusterInfo *cephclient.ClusterInfo
 	interval    time.Duration
 	client      client.Client
+	isExternal  bool
 }
 
 // newCephStatusChecker creates a new HealthChecker object
-func newCephStatusChecker(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, healthCheck cephv1.CephClusterHealthCheckSpec) *cephStatusChecker {
+func newCephStatusChecker(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, healthCheck cephv1.CephClusterHealthCheckSpec, isExternal bool) *cephStatusChecker {
```

Comment:
How about passing in the ClusterSpec? Then we don't have to keep adding additional parameters like this, and no need for the healthCheck or isExternal params.